build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -0,0 +1,259 @@
+import { Flags } from "@oclif/core";
+import { handlers, readJsonAsync } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+
+const readStdin: () => Promise<string | undefined> = async () => {
+    return new Promise((resolve) => {
+        const stdin = process.openStdin();
+        stdin.setEncoding("utf-8");
+
+        let data = "";
+        stdin.on("data", (chunk) => {
+            data += chunk;
+        });
+
+        stdin.on("end", () => {
+            resolve(data);
+        });
+
+        if (stdin.isTTY) {
+            resolve("");
+        }
+    });
+};
+
+type policyAction = "handle" | "resolve" | "final";
+
+export class CheckPolicy extends BaseCommand<typeof CheckPolicy.flags> {
+    static description =
+        "Checks that the dependencies between Fluid Framework packages are following policies.";
+
+    static flags = {
+        fix: Flags.boolean({
+            description: `Fix errors if possible`,
+            required: false,
+            char: "f",
+        }),
+        handler: Flags.string({
+            description: `Filter handler names by <regex>`,
+            required: false,
+            char: "d",
+        }),
+        path: Flags.string({
+            description: `Filter file paths by <regex>`,
+            required: false,
+            char: "p",
+        }),
+        exclusions: Flags.file({
+            description: `Path to the exclusions.json file`,
+            exists: true,
+            required: true,
+            char: "e",
+        }),
+        stdin: Flags.boolean({
+            description: `Get file from stdin`,
+            required: false,
+        }),
+        ...BaseCommand.flags,
+    };
+
+    static handlerActionPerf = new Map<policyAction, Map<string, number>>();
+    static processed = 0;
+    static count = 0;
+    static pathToGitRoot = "";
+
+    async run() {
+        if (this.processedFlags.fix) {
+            this.log("Resolving errors if possible.");
+        }
+
+        const handlerRegex: RegExp =
+            // eslint-disable-next-line no-negated-condition
+            this.processedFlags.handler !== undefined
+                ? new RegExp(this.processedFlags.handler, "i")
+                : /.?/;
+        const pathRegex: RegExp =
+            // eslint-disable-next-line no-negated-condition
+            this.processedFlags.path !== undefined
+                ? new RegExp(this.processedFlags.path, "i")
+                : /.?/;
+
+        if (this.processedFlags.handler !== undefined) {
+            this.log(`Filtering handlers by regex: ${handlerRegex}`);
+        }
+
+        if (this.processedFlags.path !== undefined) {
+            this.log(`Filtering file paths by regex: ${pathRegex}`);
+        }
+
+        if (this.processedFlags.exclusions === undefined) {
+            this.error("ERROR: No exclusions file provided.");
+        }
+
+        const exclusionsFile = await readJsonAsync(this.processedFlags.exclusions);
```

Comment:
Does readjson return a string? You might need to parse it into an object first.

Reply:
readJsonAsync seems to return a JSON object, this may throw an error. I think I should wrap it in a try catch.
code for function;
`export async function readJsonAsync(filename: string) {
    const content = await readFileAsync(filename, "utf-8");
    return JSON.parse(content);
}`